docs/hosted-orchestration/building-images.md

```diff
@@ -0,0 +1,168 @@
+
+## Building and Pushing Images
+
+### Prerequisites
+
+- A working Docker installation. For more information see [Docker's Getting Started documentation](https://docs.docker.com/get-started/).
+
+### Creating a Dockerfile
+
+To create a custom image, you must create a
+Dockerfile
+. This is a text document containing commands that Docker uses to assemble an image.
+
+### Choosing and setting a base image
+
+Before you create a custom image, you must choose another image from which to extend the custom image. Docker Hub
+has official, pre-built images for most popular languages and frameworks. Given a particular language or framework, there are many image variants from which to choose. These variants are specified by Docker tags
+
+For example, if you want to use version 3.5 of the official Alpine image, the full image name is alpine:3.5.
+
+In your Dockerfile, extend the base image by using the FROM instruction.
+
+```
+FROM golang:1.8.0
+```
+
+### Installing Additional Tools
+To install any additional tools or execute other commands, use the RUN instruction.
+
+```
+RUN apt-get update && apt-get install -y netcat
+RUN go get github.com/jstemmer/go-junit-report
+```
+
+### Adding other files and directories
+To add files and directories that are not present in package managers, use the ADD instruction.
+
+```
+ADD ./workdir/contacts /usr/bin/contacts
+ADD ./db/migrations /migrations
+```
+
+### Build Image
+After all of the required tools are specified in the Dockerfile it is possible to build the image.
+
+```
+docker build <path-to-dockerfile>
+```
+You’ll see how all commands specified in Dockerfile are executed. If there are any errors they’ll be displayed and you’ll need to fix them before continuing. If the build is successful you’ll have something like this at the very end:
+
+```
+...
+Successfully built e32703162dd4
+```
+
+Read more about docker build command.
+
+Congratulations, you’ve just built your first image! Now we need to store it somewhere to make it available for Sauce Labs.
+
+## Pushing to a Registry
+
+In order to allow Sauce Labs to use your custom image, store it in a public Docker Registry. The easiest mechanism is to create an account on Docker Hub because Docker Hub allows you to store unlimited public images for free. If your organization is already using Docker Hub you can use your existing account.
+
+Note: To use an image with the Sauce Labs Hosted Orchestration you must have a public repository. If you want to keep your image private, refer to the Using Docker Authenticated Pulls document for instructions.
+
+The example uses Docker Hub, but it is possible to use different registries if you prefer. Adapt the example based on the registry you are using.
```

Comment:
I think we should warn users that Docker Hub images, by default, are public and if they wish to keep their images secret they need a private repository (https://docs.docker.com/docker-hub/repos/#creating-a-private-repository, quay.io, gcr, etc...)

It would be a nasty surprise if a user mistakenly published their images to a public repo